homeassistant/components/sensor/rova.py

```diff
@@ -0,0 +1,150 @@
+                                 DEVICE_CLASS_TIMESTAMP)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['rova==0.0.2']
+
+# Config for rova requests.
+CONF_ZIP_CODE = 'zip_code'
+CONF_HOUSE_NUMBER = 'house_number'
+
+UPDATE_DELAY = timedelta(hours=12)
+SCAN_INTERVAL = timedelta(hours=12)
+
+# Supported sensor types:
+# Key: [json_key, name, icon]
+SENSOR_TYPES = {
+    'bio': ['gft', 'Biowaste', 'mdi:recycle'],
+    'paper': ['papier', 'Paper', 'mdi:recycle'],
+    'plastic': ['plasticplus', 'PET', 'mdi:recycle'],
+    'residual': ['rest', 'Residual', 'mdi:recycle']}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_ZIP_CODE): cv.string,
+    vol.Required(CONF_HOUSE_NUMBER): cv.string,
+    vol.Optional(CONF_NAME, default='Rova'): cv.string,
+    vol.Optional(CONF_MONITORED_CONDITIONS, default=['bio']):
+    vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)])
+})
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Create the Rova data service and sensors."""
+    from rova.rova import Rova
+    from requests.exceptions import HTTPError, ConnectTimeout
+
+    zip_code = config[CONF_ZIP_CODE]
+    house_number = config[CONF_HOUSE_NUMBER]
+    platform_name = config[CONF_NAME]
+
+    # Create new Rova object to  retrieve data
+    api = Rova(zip_code, house_number)
+
+    try:
+        if not api.is_rova_area():
+            _LOGGER.error("ROVA does not collect garbage in this area")
+            return
+    except (ConnectTimeout, HTTPError):
+        _LOGGER.error("Could not retrieve details from ROVA API")
+        return
+
+    # Create rova data service which will retrieve and update the data.
+    data_service = RovaData(api)
+
+    # Create a new sensor for each garbage type.
+    entities = []
+    for sensor_key in config[CONF_MONITORED_CONDITIONS]:
+        sensor = RovaSensor(platform_name, sensor_key, data_service)
+        entities.append(sensor)
+
+    add_entities(entities, True)
+
+
+class RovaSensor(Entity):
+    """Representation of a Rova sensor."""
+
+    def __init__(self, platform_name, sensor_key, data_service):
+        """Initialize the sensor."""
+        self.sensor_key = sensor_key
+        self.platform_name = platform_name
+        self.data_service = data_service
+
+        self._device_class = DEVICE_CLASS_TIMESTAMP
+        self._state = None
+
+        self._json_key = SENSOR_TYPES[self.sensor_key][0]
+
+    @property
+    def name(self):
+        """Return the name."""
+        return "{}_{}".format(self.platform_name, self.sensor_key)
+
+    @property
+    def icon(self):
+        """Return the sensor icon."""
+        return SENSOR_TYPES[self.sensor_key][2]
+
+    @property
+    def device_class(self):
+        """Return the class of this sensor."""
+        return self._device_class
```

Comment:
We can return the constant here directly.

Reply:
Got it, updated and pushed - thanks.